crates/matrix-sdk-ui/tests/integration/timeline/pinned_event.rs

```diff
@@ -402,3 +682,28 @@ impl TestHelper {
         self.client.sync_once(self.sync_settings.clone()).await
     }
 }
+
+enum MessageType {
+    New { id: OwnedEventId, txt: String },
+    Edit { id: OwnedEventId, old_id: OwnedEventId, txt: String },
+    Redaction { id: OwnedEventId, old_id: OwnedEventId },
+}
```

Comment:
Instead of introducing all of this, could the caller make use of an `EventFactory` immediately, and not pass a `MessageType` etc.? This would make the tests much more straightforward to read…

Reply:
That means inlining the whole `setup_sync_response` code for every sync. I honestly think that's way less readable, and it's the main reason why I abstracted it inside `TestHelper` in a previous PR.

Reply:
Can you pass the `SyncTimelineEvent`s or `TimelineEvent`s to `setup_sync_responses`?

Reply:
It's done that way now.